pkgs/development/python-modules/hikari-python/default.nix

```diff
@@ -0,0 +1,73 @@
+{ lib
+, buildPythonPackage
+, fetchFromGitHub
+, pytestCheckHook
+, pytest-runner
+, aiohttp
+, attrs
+, multidict
+, colorlog
+, pynacl
+, git
+, pytest
+, pytest-cov
+, pytest-randomly
+, pytest-asyncio
+, mock
+  # Opts
+, enableRestServer ? false
+}:
+
+buildPythonPackage rec {
+  pname = "hikari-python";
+  version = "2.0.0.dev119";
+
+  src = fetchFromGitHub {
+    owner = "hikari-py";
+    repo = "hikari";
+    rev = version;
+    sha256 = "sha256-4uMY0NZaLxl52Wl471xQ6JqWWVW5QzMOyrXeVaZltfo=";
+    leaveDotGit = true;
+  };
+
+  patchInputs = [
+    git
+  ];
+
+  propagatedBuildInputs = [
+    aiohttp
+    attrs
+    multidict
+    colorlog
+  ] ++ lib.optional enableRestServer [ pynacl ];
+
+  checkInputs = [
+    pytest
+    pytest-asyncio
+    pytest-cov
+    pytest-randomly
+    mock
+  ];
+
+  nativeCheckInputs = [
+    pytestCheckHook
+    pytest-runner
+  ];
+
+  postPatch = ''
+    # Hikari wants a Git commit hash.
+    export SHA=$(${git}/bin/git log -n 1 --pretty=format:"%H")
+    substituteInPlace hikari/_about.py --replace "__git_sha1__.* = (.*)" $SHA
+
+    echo "" > requirements.txt
```

Comment:
What is the reason for emptying requirements.txt?

Reply:
Version constraints were acting oddly with Nixpkgs.

Reply:
@Janik-Haag should I not do this?

Reply:
To be honest I'm not quite sure.

Reply:
We can use `pythonRelaxDepsHooks` here.

Reply:
Updated to use `pythonRelaxDepsHook`.